pkgs/top-level/python-packages.nix

```diff
@@ -4196,6 +4196,28 @@ in modules // {
       description = "pytest plugin with mechanisms for caching across test runs";
     };
   };
+  
+  pytestdjango = buildPythonPackage rec {
+    name = "pytest-django-${version}";
+    version = "2.9.1";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/p/pytest-django/${name}.tar.gz";
+      sha256 = "1mmc7zsz3dlhs6sx4sppkj1vgshabi362r1a8b8wpj1qfximpqcb";
+    };
+    
+    # doing this to allow depending packages to find
+    # pytest's binaries
+    pytest = self.pytest;
```

Comment:
Perhaps what you are looking for is like for example `numpy.blas` where in the function that builds `numpy` we have passthru.blas = blas`.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/numpy.nix

That way, if another package needs that version of `pytest` as a `buildInput`, they can just add `pytestdjango.pytest`.

I don't think that's necessary here since in `python-packages.nix` we have only one `pytest`. The reason it is done for `numpy.blas` is that in `all-packages.nix` we have different BLAS implementations, and in case one of the Python packages uses a different one than the other we could get segmentation faults.